src/haystack/analyzer.clj

```diff
@@ -338,14 +338,25 @@
           (flag-duplicates)
           (flag-tooling)))))
 
+(defn- compile-like-exception?
+  [{cause-type    :type
+    ^String
+    cause-message :message}]
+  (and (= cause-type 'java.lang.IllegalArgumentException)
+       (or (some-> cause-message (.startsWith "No matching field"))
+           (some-> cause-message (.startsWith "No matching method")))))
+
 (defn- analyze-cause
   "Analyze the `cause-data` of an exception, in `Throwable->map` format."
   [cause-data print-fn]
   (let [pprint-str #(let [writer (StringWriter.)]
                       (print-fn % writer)
                       (str writer))
+        phase (-> cause-data :data :clojure.error/phase)
         m {:class (name (:type cause-data))
-           :phase (-> cause-data :data :clojure.error/phase)
+           :phase phase
```

Comment:
At first I was tempted to put a synthetic value under `:phase`, but that might be surprising to anyone using Haystack as a lib.

I plan to do some transparent changes in cider.el such that we can we can change this default:

```diff
(defcustom cider-clojure-compilation-error-phases '("read-source"
                                                    "macro-syntax-check"
                                                    "macroexpansion"
                                                    "compile-syntax-check"
                                                    "compilation"
+                                                   "cider/compile-like"
```